File: SAMSUI/src/app/Model/master/asset-group-dtl.ts
import { Model } from "./model";
import { AssetCustFieldValue } from "./asset-cust-field-value";
import { BaseModel } from "../base/baseModel";
import { AssetGroupModel } from "./asset-group";

export class AssetGroupDtlModel extends BaseModel {

    public assetGroupDtlId ?: number;

    public assetGroupId ?: number;

    public modelId ?: number;

    public modelName ?: string;
    
    public active ?:boolean;

    public createdBy ?:string;

    public createdDtDisp ?: string;

    public updatedBy ?: string;

    public updatedDtDisp ?: string;

    public pageNumber ?: number;

    public recordsPerPage ?: number;

    public modelTO ?: Model;

    public assetGroupTO ?: AssetGroupModel;

    

}

File: sams-backend/src/main/java/com/sams/entity/AssetGroupDtl.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_asset_group_dtl", schema = "asset")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AssetGroupDtl {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "asset_group_dtl_id")
    private Long assetGroupDtlId;

    @Column(name = "asset_group_id")
    private Long assetGroupId;

    @Column(name = "model_id")
    private Long modelId;

    @Column(name = "model_name")
    private String modelName;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "model_t_o")
    private String modelTO;

    @Column(name = "asset_group_t_o")
    private String assetGroupTO;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}